### PR TITLE
Rename URL to ShortLink in usecase directory

### PR DESCRIPTION
**Pull request by arberiii**
Part of [#794](https://github.com/short-d/short/issues/794). This large PR renames all appearance of URL in directory short-d/short/backend/app/usecase to ShortLink accordingly, except for a TODO which I think is best to leave it as the issue. Sorry for the large PR! There is still the appearance of URL in other directories but I intend to change them in their respective PRs.

### CODE REVIEW

File: backend/app/usecase/shortlink/shortlinkcreator_test.go [hunk 1]
```diff
@@ -157,12 +157,12 @@ func TestURLCreatorPersist_CreateURL(t *testing.T) {
 		t.Run(testCase.name, func(t *testing.T) {
 			t.Parallel()
 
-			blockedURLs := map[string]bool{}
-			blacklist := risk.NewBlackListFake(blockedURLs)
-			shortLinkRepo := repository.NewShortLinkFake(testCase.urls)
+			blockedShortLinks := map[string]bool{}
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkcreator_test.go:160]:
```suggestion
			blockedURLs := map[string]bool{}
```

Comment (byliuyang) [reply backend/app/usecase/shortlink/shortlinkcreator_test.go:160]: This is real URL.

File: backend/app/usecase/shortlink/shortlinkretriever.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package shortlink
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkretriever.go:1]: Please rename the file to retriever. shortlink is already in package name.

File: backend/app/usecase/shortlink/shortlinkretriever_test.go [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // +build !integration all
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkretriever_test.go:1]: Please rename this file to `retriever_test.go`.

File: backend/app/usecase/shortlink/shortlinkretriever_test.go [hunk 4]
```diff
@@ -11,47 +11,47 @@ import (
...
+			shortLinks:        shortLinks{},
+			alias:             "220uFicCJj",
+			expiringAt:        &now,
+			hasErr:            true,
+			expectedShortLink: entity.ShortLink{},
 		},
 		{
-			name: "url expired",
-			urls: urlMap{
+			name: "shortlink expired",
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkretriever_test.go:40]:
```suggestion
			name: "short link expired",
```

File: backend/app/usecase/shortlink/shortlinkretriever_test.go [hunk 5]
```diff
@@ -11,47 +11,47 @@ import (
...
-			expectedURL: entity.ShortLink{},
+			alias:             "220uFicCJj",
+			expiringAt:        &now,
+			hasErr:            true,
+			expectedShortLink: entity.ShortLink{},
 		},
 		{
-			name: "url never expire",
-			urls: urlMap{
+			name: "shortlink never expire",
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkretriever_test.go:53]:
```suggestion
			name: "short link never expire",
```

File: backend/app/usecase/shortlink/shortlinkretriever_test.go [hunk 6]
```diff
@@ -60,14 +60,14 @@ func TestUrlRetriever_GetURL(t *testing.T) {
...
-			expectedURL: entity.ShortLink{
+			expectedShortLink: entity.ShortLink{
 				Alias:    "220uFicCJj",
 				ExpireAt: nil,
 			},
 		},
 		{
-			name: "unexpired url found",
-			urls: urlMap{
+			name: "unexpired shortlink found",
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkretriever_test.go:69]:
```suggestion
			name: "unexpired short link found",
```

File: backend/app/usecase/shortlink/shortlinkretriever_test.go [hunk 7]
```diff
@@ -11,47 +11,47 @@ import (
 	"github.com/short-d/short/backend/app/usecase/repository"
 )
 
-type urlMap = map[string]entity.ShortLink
+type shortLinks = map[string]entity.ShortLink
 
-func TestUrlRetriever_GetURL(t *testing.T) {
+func TestShortLinkRetriever_GetShortLink(t *testing.T) {
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkretriever_test.go:16]:
```suggestion
func TestRetriever_GetShortLink(t *testing.T) {
```

File: backend/app/usecase/shortlink/shortlinkretriever_test.go [hunk 8]
```diff
@@ -88,36 +88,36 @@ func TestUrlRetriever_GetURL(t *testing.T) {
...
+		users              []entity.User
+		createdShortLinks  []entity.ShortLink
+		user               entity.User
+		hasErr             bool
+		expectedShortLinks []entity.ShortLink
 	}{
 		{
-			name: "user created URLs",
-			urls: urlMap{
+			name: "user created ShortLinks",
```
Comment (byliuyang) [inline backend/app/usecase/shortlink/shortlinkretriever_test.go:119]:
```suggestion
			name: "user created short link",
```

File: backend/app/usecase/sso/identityprovider.go [hunk 9]
```diff
@@ -3,6 +3,6 @@ package sso
 // IdentityProvider represents external service that verifies the user's
 // identity.
 type IdentityProvider interface {
-	GetAuthorizationURL() string
+	GetAuthorizationShortLink() string
```
Comment (byliuyang) [inline backend/app/usecase/sso/identityprovider.go:6]: This is real URL. Please change it back.

File: backend/app/usecase/sso/identityprovider_fake.go [hunk 10]
```diff
@@ -5,14 +5,14 @@ var _ IdentityProvider = (*IdentityProviderFake)(nil)
 // IdentityProviderFake represents in memory implementation of an external
```
Comment (byliuyang) [inline backend/app/usecase/sso/identityprovider_fake.go:5]: This file is about real URL of the sign in link. Please revert the changes.

File: backend/app/usecase/sso/sso.go [hunk 11]
```diff
@@ -58,7 +58,7 @@ func (o SingleSignOn) IsSignedIn(authToken string) bool {
 
 // GetSignInLink retrieves the sign in link of the external account provider.
 func (o SingleSignOn) GetSignInLink() string {
-	return o.identityProvider.GetAuthorizationURL()
+	return o.identityProvider.GetAuthorizationShortLink()
```
Comment (byliuyang) [inline backend/app/usecase/sso/sso.go:61]: This represents sign in URL. Please revert the changes.

File: backend/dep/provider/routes.go [hunk 12]
```diff
@@ -20,7 +20,7 @@ func NewShortRoutes(
 	instrumentationFactory request.InstrumentationFactory,
 	webFrontendURL WebFrontendURL,
 	timer timer.Timer,
-	urlRetriever url.Retriever,
+	urlRetriever shortlink.Retriever,
```
Comment (byliuyang) [inline backend/dep/provider/routes.go:23]:
```suggestion
	shortLinkRetriever shortlink.Retriever,
```

File: backend/app/adapter/gqlapi/resolver/authquery_test.go [hunk 13]
```diff
@@ -31,15 +31,15 @@ func TestAuthQuery_URL(t *testing.T) {
 		user        entity.User
 		alias       string
 		expireAfter *scalar.Time
-		urls        urlMap
+		urls        shortLinkMap
```
Comment (byliuyang) [inline backend/app/adapter/gqlapi/resolver/authquery_test.go:34]:
```suggestion
		shortLinks        shortLinkMap
```

File: backend/app/usecase/risk/blacklist.go [hunk 14]
```diff
@@ -2,5 +2,5 @@ package risk
 
 // BlackList checks whether an item is acceptable
 type BlackList interface {
-	HasURL(url string) (bool, error)
+	HasURL(shortLink string) (bool, error)
```
Comment (byliuyang) [inline backend/app/usecase/risk/blacklist.go:5]:
```suggestion
	HasURL(url string) (bool, error)
```

Comment (arberiii) [reply backend/app/usecase/risk/blacklist.go:5]: Thank you for through review!
